### DISCUSSION TITLE
smp-status API certificate

### DISCUSSION THREAD

**Discussion by jonaswest** [phax/phoss-smp#172] [Q&A]
Hi,

We are currently discussing how we should monitor whether the certificate we have is expired or not. Today, with the help of the smp-status page, you can see this through smp.certificate.expired, but we would like to be able to see which date it applies so you can automatically set up an alarm on it. Can you somehow get smp.certificate.expiry.date through api, for example?

Best Regards
Jonas

**Comment by phax** [comment]
Hi Jonas,
Thanks for reaching out to me. Well yes in general, that is technically feasible, but I am not sure if it is okay, to publish that data by default (because based on the duration of the expiration you may be able to deduce some attack vector or so????).
So my proposal would be to add a new configuration option (something in the form `smp.status.certificate.details`) with a default value of `false`, that would enable the inclusion of this date.
If that sounds reasonable to you, it would be create if you could create a respective issue for this.
hth, Philip

**Reply by phax** [reply]
@jonaswest  I created an issue out of this and will up on it soon

**Reply by phax** [reply]
It will be part of the v5.7.0 release - Use `smp.status.show.certificate.dates` to enable the feature in your configuration
